apps/meteor/server/services/messages/hooks/BeforeSaveJumpToMessage.ts

```diff
@@ -71,6 +71,10 @@ export class BeforeSaveJumpToMessage {
 			useRealName: boolean;
 		};
 	}): Promise<IMessage> {
+		if (message?.attachments) {
+			// Do not keep old quote attachments since they may not still be linked to the message
+			message.attachments = message.attachments?.filter((attachment) => !isQuoteAttachment(attachment));
```

Comment:
In my opinion, it would be clearer if we could create a private function with a proper name for this operation. 

From what I understood we are "resetting" the attachments at this point, removing all the quotes and keeping the other kind of attachments, right? So, if we have new URLs(quotes) to parse it will reparse below this point, otherwise(if the existing quotes were deleted) it will return the message without the old (quote) attachment.

What do you think?